scripts/prepare_sector_network.py

```diff
@@ -3100,14 +3088,15 @@ def add_biomass(n, costs):
                 )
                 * average_distance,
             )
-            # Set last snapshot of e_max_pu for unsustainable solid biomass to 1 to make operational limit work
-            unsus_stores_idx = n.stores.query(
-                "carrier == 'unsustainable solid biomass'"
-            ).index
-            unsus_stores_idx = unsus_stores_idx.intersection(
-                n.stores_t.e_max_pu.columns
-            )
-            n.stores_t.e_max_pu.loc[n.snapshots[-1], unsus_stores_idx] = 1
+            # Set e_sum_min to 0 to allow for the faux biomass transport
+            n.generators.loc[
+                n.generators.carrier == "unsustainable solid biomass", "e_sum_min"
+            ] = 0
+            # Set e_sum_max to the potential to limit the faux biomass transport
+            n.generators.loc[
+                n.generators.carrier == "unsustainable solid biomass", "e_sum_max"
+            ] = unsustainable_solid_biomass_potentials_spatial.sum()
```

Comment:
Not sure if this is the right constraint. There is one generator for every bus, and if every bus has access to the sum of the total potential wouldn't we get `n_buses * total_potential`  of import capacity?

Reply:
ah i see, the GlobalConstraint is just below and remains there. Then the questions if, if this e_sum_max constraint is still needed or only a duplicate?

Reply:
I think you are right that it is not only redundant, but actually false. The e_sum_max set at the original generator intialization in https://github.com/PyPSA/pypsa-eur/blob/42c262eeef2380f48c550000b4d2dc949db5137f/scripts/prepare_sector_network.py#L2913 should remain. All the generation needed beyond that local limit must be provided by the other "artifical" generator, that also takes into account the transport costs.